lib/bigtable/family.js

```diff
@@ -0,0 +1,257 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /*!
+ * @module bigtable/family
+ */
+
+'use strict';
+
+var nodeutil = require('util');
+var is = require('is');
+
+/**
+ * @type {module:common/grpcServiceObject}
+ * @private
+ */
+var GrpcServiceObject = require('../common/grpc-service-object.js');
+
+/**
+ * Create a Family object to interact with your table column families.
+ *
+ * @constructor
+ * @alias module:bigtable/family
+ *
+ * @example
+ * var gcloud = require('gcloud');
+ *
+ * var bigtable = gcloud.bigtable({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123',
+ *   cluster: 'zones/us-central1-b/clusters/gcloud-node'
+ * });
+ *
+ * var table = bigtable.table('presidents');
+ * var family = table.family('name');
+ */
+function Family(table, name) {
+  this.name = Family.formatName_(table, name);
+
+  var methods = {
+
+    /**
+     * Create a column family.
+     *
+     * @param {object=} options - See {module:bigtable/table#createFamily}.
+     *
+     * @example
+     * family.create(function(err, family, apiResponse) {
+     *   // The column family was created successfully.
+     * });
+     */
+    create: true,
+
+    /**
+     * Delete the column family.
+     *
+     * @param {function=} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {object} callback.apiResponse - The full API response.
+     *
+     * @example
+     * family.delete(function(err, apiResponse));
+     */
+    delete: {
+      protoOpts: {
+        service: 'BigtableTableService',
+        method: 'deleteColumnFamily'
+      },
+      reqOpts: {
+        name: this.name
+      }
+    },
+
+    /**
+     * Check if the column family exists.
+     *
+     * @param {function} callback - The callback function.
+     * @param {?error} callback.err - An error returned while making this
+     *     request.
+     * @param {boolean} callback.exists - Whether the family exists or not.
+     *
+     * @example
+     * family.exists(function(err, exists) {});
+     */
+    exists: true,
+
+    /**
+     * Get a column family if it exists.
+     *
+     * You may optionally use this to "get or create" an object by providing an
+     * object with `autoCreate` set to `true`. Any extra configuration that is
+     * normally required for the `create` method must be contained within this
+     * object as well.
+     *
+     * @param {options=} options - Configuration object.
+     * @param {boolean} options.autoCreate - Automatically create the object if
+     *     it does not exist. Default: `false`
+     *
+     * @example
+     * family.get(function(err, family, apiResponse) {
+     *   // The `family` data has been updated.
```

Comment:
The pattern with other `get` methods has a comment like:

```
// `dataset.metadata` has been populated.
```

It gives a clue that the user should check the `metadata` property for the latest state from the API. Can we elaborate here to give the same hint to the reader?